mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/HighAvailabilityServicesUtil.java

```diff
@@ -121,99 +124,103 @@ protected void startUp() throws Exception {
+            final String metricsGroup = masterMonitor instanceof ZookeeperMasterMonitor ? "ZkHighAvailabilityServices" :
+                "HighAvailabilityServices";
+
+            final Metrics metrics = MetricsRegistry.getInstance().registerAndGet(new Metrics.Builder()
+                .name(metricsGroup)
+                .addCounter("resourceLeaderChangeCounter")
+                .addCounter("resourceLeaderAlreadyRegisteredCounter")
+                .build());
+            resourceLeaderChangeCounter = metrics.getCounter("resourceLeaderChangeCounter");
+            resourceLeaderAlreadyRegisteredCounter = metrics.getCounter("resourceLeaderAlreadyRegisteredCounter");
 
-  /**
-   * Zookeeper based implementation of HighAvailabilityServices that finds the various leader instances
-   * through metadata stored on zookeeper.
-   */
-  private static class ZkHighAvailabilityServices extends AbstractIdleService implements
-      HighAvailabilityServices {
-
-    private final CuratorService curatorService;
-    private final Counter resourceLeaderChangeCounter;
-    private final Counter resourceLeaderAlreadyRegisteredCounter;
-    private final AtomicInteger rmConnections = new AtomicInteger(0);
-    private final CoreConfiguration configuration;
-
-    public ZkHighAvailabilityServices(CoreConfiguration configuration) {
-      curatorService = new CuratorService(configuration);
-      final Metrics metrics = MetricsRegistry.getInstance().registerAndGet(new Metrics.Builder()
-        .name("ZkHighAvailabilityServices")
-        .addCounter("resourceLeaderChangeCounter")
-        .addCounter("resourceLeaderAlreadyRegisteredCounter")
-        .build());
-      resourceLeaderChangeCounter = metrics.getCounter("resourceLeaderChangeCounter");
-      resourceLeaderAlreadyRegisteredCounter = metrics.getCounter("resourceLeaderAlreadyRegisteredCounter");
-      this.configuration = configuration;
-    }
-
-    @Override
-    protected void startUp() throws Exception {
-      curatorService.start();
-    }
-
-    @Override
-    protected void shutDown() throws Exception {
-      curatorService.shutdown();
-    }
-
-    @Override
-    public MantisMasterGateway getMasterClientApi() {
-      return new MantisMasterClientApi(curatorService.getMasterMonitor());
-    }
-
-    @Override
-    public MasterMonitor getMasterMonitor() {
-      return curatorService.getMasterMonitor();
-    }
-
-    @Override
-    public ResourceLeaderConnection<ResourceClusterGateway> connectWithResourceManager(
-        ClusterID clusterID) {
-      return new ResourceLeaderConnection<ResourceClusterGateway>() {
-        final MasterMonitor masterMonitor = curatorService.getMasterMonitor();
+        }
 
-        ResourceClusterGateway currentResourceClusterGateway =
-            new ResourceClusterGatewayClient(clusterID, masterMonitor.getLatestMaster(), configuration);
```

Comment:
This part of the file is a confusing unified diff, but this is moved not deleted. If you find changes here please flag them as the intent was no logic code change, just moving.